### ISSUE TITLE
Error with integer design variables

### ISSUE THREAD

**Issue by pjstanle** [mdolab/pyoptsparse#163]
## Type of issue
What types of issue is it?
- Bugfix (non-breaking change which fixes an issue)

## Description
Unable to optimize with integer design variables.
### Steps to reproduce issue

```

from pyoptsparse import Optimization, NSGA2

def obj_func(xdict):

    x = xdict['x']
    y = xdict['y']

    funcs = {}

    funcs['obj'] = (2+x*y) + x**2 - 5*y

    fail = False

    return funcs, fail

## Main
if __name__ == "__main__":

    x0 = 3
    y0 = 1    

    # """Optimization"""
    optProb = Optimization('integer_example', obj_func)
    optProb.addObj('obj')

    optProb.addVar('x', 'i', lower=-10, upper=10)
    optProb.addVar('y', 'i', lower=-10, upper=10)

    opt = NSGA2()
    opt.setOption('PopSize', 5)
    opt.setOption('maxGen', 10)


    res = opt(optProb)

    xf = res.xStar['x']
    yf = res.xStar['y']

    print("xf: ", xf)
    print("yf: ", yf)
```

### Current behavior
```
Traceback (most recent call last):
  File "integer_example.py", line 35, in <module>
    res = opt(optProb)
  File "/Users/astanley/Programs/miniconda3/lib/python3.7/site-packages/pyoptsparse/pyNSGA2/pyNSGA2.py", line 118, in __call__
    self.optProb.finalizeConstraints()
  File "/Users/astanley/Programs/miniconda3/lib/python3.7/site-packages/pyoptsparse/pyOpt_optimization.py", line 874, in finalizeConstraints
    xoffset.append(var.offset)
AttributeError: 'Variable' object has no attribute 'offset'
```

### Code version (if relevant)
**Python version:** 3.7.6
**pyoptsparse version:** 2.1.7

**Comment by nwu63** [comment]
Labelling as wontfix since pyOptSparse does not currently ship with any optimizers capable of working with discrete variables. This was something that was supported by the original pyOpt package but pyOptSparse is designed and maintained for the purposes of large-scale gradient-based optimization, so those types of things are going to be poorly supported as a result.

**Comment by nwu63** [comment]
Closed by #192.
